pico_project.py

```diff
@@ -1128,127 +1128,134 @@ def generateProjectFiles(projectPath, projectName, sdkPath, projects, debugger):
-                   '    "CMAKE_MODULE_PATH": "${env:PICO_INSTALL_PATH}/pico-sdk-tools"\n'
-                   '  },\n'
-                   '  "cmake.generator": "Ninja",\n'
-                   '  "C_Cpp.default.configurationProvider": "ms-vscode.cmake-tools"\n'
-                   '}\n')
-
-            extensions = ( '{\n'
-                   '  "recommendations": [\n'
-                   '    "marus25.cortex-debug",\n'
-                   '    "ms-vscode.cmake-tools",\n'
-                   '    "ms-vscode.cpptools",\n'
-                   '    "ms-vscode.cpptools-extension-pack",\n'
-                   '    "ms-vscode.vscode-serial-monitor"\n'
-                   '  ]\n'
-                   '}\n')
+            launch = f'''{{
+    "version": "0.2.0",
+    "configurations": [
+        {{
+            "name": "Pico Debug (Cortex-Debug)",
+            "cwd": "${{workspaceRoot}}",
+            "executable": "${{command:cmake.launchTargetPath}}",
+            "request": "launch",
+            "type": "cortex-debug",
+            "servertype": "openocd",
+            "gdbPath": "{gdbPath}",
+            "device": "RP2040",
+            "configFiles": [
+                "interface/{debugger}",
+                "target/rp2040.cfg"
+            ],
+            "svdFile": "${{env:PICO_SDK_PATH}}/src/rp2040/hardware_regs/rp2040.svd",
+            "runToEntryPoint": "main",
+            // Give restart the same functionality as runToEntryPoint - main
+            "postRestartCommands": [
+                "break main",
+                "continue"
+            ],
+            "openOCDLaunchCommands": [
+                "adapter speed 1000"
+            ]
+        }},
+        {{
+            "name": "Pico Debug (Cortex-Debug with external OpenOCD)",
+            "cwd": "${{workspaceRoot}}",
+            "executable": "${{command:cmake.launchTargetPath}}",
+            "request": "launch",
+            "type": "cortex-debug",
+            "servertype": "external",
+            "gdbTarget": "localhost:3333",
+            "gdbPath": "{gdbPath}",
+            "device": "RP2040",
+            "svdFile": "${{env:PICO_SDK_PATH}}/src/rp2040/hardware_regs/rp2040.svd",
+            "runToEntryPoint": "main",
+            // Give restart the same functionality as runToEntryPoint - main
+            "postRestartCommands": [
+                "break main",
+                "continue"
+            ]
+        }},
+        {{
+            "name": "Pico Debug (C++ Debugger)",
+            "type": "cppdbg",
+            "request": "launch",
+            "cwd": "${{workspaceRoot}}",
+            "program": "${{command:cmake.launchTargetPath}}",
+            "MIMode": "gdb",
+            "miDebuggerPath": "{gdbPath}",
+            "miDebuggerServerAddress": "localhost:3333",
+            "debugServerPath": "openocd",
+            "debugServerArgs": "-f interface/cmsis-dap.cfg -f target/rp2040.cfg -c \\"adapter speed 1000\\"",
```

Comment:
ping @P33M  - should this be "adapter speed 5000" rather than "adapter speed 1000" ?

Reply:
I believe 5000 is the current recommendation, that may have changed since I was told though.

Reply:
> ping @P33M - should this be "adapter speed 5000" rather than "adapter speed 1000" ?

yes

Reply:
Fixed in https://github.com/raspberrypi/pico-project-generator/pull/91